agreement/player.go

```diff
@@ -285,9 +285,17 @@ func (p *player) updateCredentialArrivalHistory(r routerHandle, ver protocol.Con
 	if p.Period != 0 {
```

Comment:
This check doesn't quite make sense: we're checking the period of the player in `p.Round` but then this affects whether we fetch the proposal arrival time for `p.Round - credentialRoundLag`.

Reply:
You're right. It's now redundant.
If the node enters a non-0 period, it'll discard the history we keep and start over. 
This also means that credentialRoundLag from that period, it will try and fail, to read the propagation time from period 0 from credentialRoundLag rounds ago. I think this is okay since it'll just skip the measurement from this round.